pkg/kv/kvserver/raft_apply_asserter.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2019 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+// TODO: put in a test package
+package kvserver
```

Comment:
Consider putting this into the `kvserver/apply` package if dependencies allow. This effectively codifies the contract.

Reply:
Yeah, good idea.